servers/rendering/rendering_device.h

```diff
@@ -396,8 +396,8 @@ class RenderingDevice : public Object {
 		BARRIER_MASK_RASTER = 1,
 		BARRIER_MASK_COMPUTE = 2,
 		BARRIER_MASK_TRANSFER = 4,
+		BARRIER_MASK_ALL = BARRIER_MASK_RASTER | BARRIER_MASK_COMPUTE | BARRIER_MASK_TRANSFER, // 7
```

Comment:
Not blocking for this PR but it seems a bit weird to have an `BARRIER_MASK_ALL` enum value that doesn't actually include all values.

Maybe `BARRIER_MASK_NO_BARRIER = 8` should be `BARRIER_MASK_NONE = 0` instead? CC @clayjohn @BastiaanOlij

Reply:
Ah, there is some lines in code which are checking for this like:
```
if (p_barrier != RD::BARRIER_MASK_NO_BARRIER) {
	RD::get_singleton()->barrier(RD::BARRIER_MASK_RASTER, p_barrier);
}
```
So if we are to change it to 0 this code will not execute when default 0 have passed.

Reply:
Or we could unexposed `BARRIER_MASK_ALL` since it's just a combination of the previous values...

Reply:
Well I'm not familiar with how this API is used so this was a suggestion to assess in the rendering team, I don't know if it is actually a good idea or not.

Reply:
Yes, I leave comments for the rendering team ^^

I think the following structure may be better (exposing `BARRIER_MASK_ALL` as a constant and other as bitfield flags):

```
	enum BarrierMask {
		BARRIER_MASK_RASTER = 1,
		BARRIER_MASK_COMPUTE = 2,
		BARRIER_MASK_TRANSFER = 4,
		BARRIER_MASK_NO_BARRIER = 8,
	};

	static const int BARRIER_MASK_ALL = BARRIER_MASK_RASTER | BARRIER_MASK_COMPUTE | BARRIER_MASK_TRANSFER;
```